python/lsst/meas/astrom/matchOptimisticB/matchOptimisticB.cc

```diff
@@ -102,7 +102,7 @@ PYBIND11_MODULE(matchOptimisticB, mod) {
             (ProxyVector(*)(afw::table::SimpleCatalog const &, afw::geom::SkyWcs const &)) & makeProxies,
             "posRefCat"_a, "tanWcs"_a);
 
-    mod.def("matchOptimisticB", &matchOptimisticB, "posRefCat"_a, "sourceCat"_a, "control"_a, "wcs"_a,
+    mod.def("matchCatalogsOptimisticB", &matchOptimisticB, "posRefCat"_a, "sourceCat"_a, "control"_a, "wcs"_a,
```

Comment:
I am puzzled why you are making this particular change. I _think_ it would be preferable to leave the name alone and stop hoisting the `matchOptimisticB` function up a level (meaning it can only be referenced by `lsst.meas.astrom.matchOptimisticB.matchOptimisticB`). The other public symbols can still be hoisted, as you are doing, leaving most code working without any changes to the C++.

I am concerned that renaming this function will be very confusing because it doesn't match the C++ function, nor the name of the pybind11 file, nor the name of the task. I really hope you can avoid this rename.